compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1033,6 +1037,17 @@ rustc_queries! {
         cache_on_disk_if { true }
     }
 
+    /// A list of types where the ADT requires drop if and only if any of those
+    /// types has significant drop. A drop is significant if it is implemented
```

Comment:
This sort of gives an illusion that you will detect regardless of if the rustc_ins... is present, and that the attribute is helper for certain cases.